### PR TITLE
Spcqe quantiles

### PR DESCRIPTION
**Pull request by AramisDuf**
Implementing time dilatation of pv signal + fitting quantiles

### CODE REVIEW

File: solardatatools/algorithms/interpolation.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    cumulative_integrals = np.zeros(ttnew.shape[0])
+    # Add the initial zero as a baseline for the cumulative sum
+    cumulative_integrals[1:] = np.cumsum(piecewise_integrals)
+    was_nan = np.isnan(xxnew)
+    was_nan[-1] = False # the last point is not used in the interpolation
+    cumulative_integrals[was_nan] = np.nan # set NaNs back to NaN
+    y = np.diff(cumulative_integrals[new_indices])
+    return y
+
+def interpolate(tnew, t, x, alignment='left'):
```
Comment (AramisDuf) [inline solardatatools/algorithms/interpolation.py:49]: Not sphinx docstring

File: solardatatools/algorithms/interpolation.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        alignment (str): Time bin label alignement, either 'left' or 'right'.
+
+    Returns:
+        y (np.ndarray): Interpolated signal for the new time points (length n-1).
+    '''
+
+    check_sanity(tnew, t, x, alignment)
+    # make sure everything is float, flip if alignment is right
+    tnew_float, t_float, x_float = initialize_arrays(tnew, t, x, alignment)
+    indices = np.searchsorted(t_float, tnew_float, side='right')
```
Comment (AramisDuf) [inline solardatatools/algorithms/interpolation.py:67]: add comments

File: solardatatools/algorithms/interpolation.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    '''
+    Parameters:
+        ttnew (np.ndarray): t + tnew (length m+n).
+        xxnew (np.ndarray): x + xnew (length m+n).
+        new_indices (np.ndarray): Indices of the points in tnew (length n).
+
+    Returns:
+        y (np.ndarray): Interpolated signal (length n-1).
+    '''
+    xxnew_filled = np.nan_to_num(xxnew, nan=0) # replace NaNs with zeros
```
Comment (AramisDuf) [inline solardatatools/algorithms/interpolation.py:38]: More comments

File: solardatatools/algorithms/interpolation.py (hunk 1 above)
Comment (AramisDuf) [inline solardatatools/algorithms/interpolation.py:49]: move it to the top

File: solardatatools/algorithms/dilatation.py [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+
+import numpy as np
+from solardatatools.plotting import plot_2d
+
+DEFAULT = {
+    "nvals_dil": 101,
+}
+
+class Dilatation:
+    def __init__(self, data_handler, **config):
```
Comment (bmeyers) [inline solardatatools/algorithms/dilatation.py:16]: we should document somewhere what can be in `config`

File: solardatatools/algorithms/dilatation.py [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+    :param signal_idx_dil: ndarray, 1D array with the dilated index (length n).
+    :param signal_idx_ori: ndarray, 1D array with the original index (length m).
+    :param signal_ori: ndarray, 1D array with the original signal values (length m-1).
+    :return: ndarray, 1D array with the dilated signal values (length n-1).
+    """
+    _signal_ori = np.append(signal_ori, signal_ori[-1]) # Adding last dummy value to interpolate
+    signal_dil = interpolate(signal_idx_dil, signal_idx_ori, _signal_ori, alignment='left')
+    return signal_dil
+
+def undilate_signal(signal_idx_ori, signal_idx_dil, signal_dil):
```
Comment (bmeyers) [inline solardatatools/algorithms/dilatation.py:102]: change to do necessary insertion of zero values as appropriate for solar data

File: solardatatools/algorithms/dilatation.py [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+    :param signal_idx_ori: ndarray, 1D array with the original index (length m).
+    :param signal_idx_dil: ndarray, 1D array with the dilated index (length n).
+    :param signal_dil: ndarray, 1D array with the dilated signal values (length n-1).
+    :return: ndarray, 1D array with the original signal values (length m-1).
+    """
+    _signal_dil = np.append(signal_dil, signal_dil[-1]) # Adding last dummy value to interpolate
+    signal_ori = interpolate(signal_idx_ori, signal_idx_dil, _signal_dil, alignment='left')
+    return signal_ori
+
+def undilate_quantiles(signal_idx_ori, signal_idx_dil, quantiles_dil, nvals_dil=DEFAULT["nvals_dil"]):
```
Comment (bmeyers) [inline solardatatools/algorithms/dilatation.py:115]: reference updated function above

**Review by bmeyers — COMMENTED** [review]
good progress

**Review by bmeyers — APPROVED** [review]

**Pull request merged**